### PR TITLE
doc: lint README.md

### PR DESCRIPTION
**Pull request by maclover7**
##### Checklist

- [x] `npm install && npm test` passes
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Description of change
Fixes grammar, removes extra lines and spaces, etc. Also removes a few
references to `node-waf`, which was removed ~6 years ago now. Happy to
add back if people still need that information.

cc docs guru @Trott

### CODE REVIEW

**Comment by Trott** [comment]
I'll in turn ping @vsemozhetbyt. :-D

File: README.md [hunk 1]
```diff
@@ -110,33 +102,30 @@ needs to be added (not needed when run by npm as configured above):
...
+you can require the `.node` file with Node.js and run your tests!
 
 __Note:__ To create a _Debug_ build of the bindings file, pass the `--debug` (or
-`-d`) switch when running either the `configure`, `build` or `rebuild` command.
-
+`-d`) switch when running either the `configure`, `build` or `rebuild` commands.
 
-The "binding.gyp" file
-----------------------
+## The "binding.gyp" file
```
Comment (vsemozhetbyt) [inline README.md]: Maybe `"binding.gyp"` -> `` `binding.gyp` ``?

File: README.md [hunk 2]
```diff
@@ -170,86 +158,74 @@ Commands
...
+$ export npm_config_devdir=/tmp/.gyp
+```
+
+Or this on Windows:
+
+```console
+> set npm_config_devdir=c:\temp\.gyp
+```
+
+### npm configuration
```
Comment (vsemozhetbyt) [inline README.md]: `npm` -> `` `npm` ``?

File: README.md [hunk 3]
```diff
@@ -170,86 +158,74 @@ Commands
...
+$ npm config set [--global] devdir /tmp/.gyp
+```
+
+**Note:** Configuration set via `npm` will only be used when `node-gyp`
+is run via `npm`, not when `node-gyp` is run directly.
+
+## License
+
+`node-gyp` is available under the MIT license. See the [LICENSE
+file][LICENSE] for details.
```
Comment (vsemozhetbyt) [inline README.md]: `[LICENSE]` -> `(LICENSE)` (square brackets need a bottom reference, otherwise the link is not rendered).

File: README.md [hunk 4]
```diff
@@ -110,33 +102,30 @@ needs to be added (not needed when run by npm as configured above):
...
-Previously when node had `node-waf` you had to write a `wscript` file. The
-replacement for that is the `binding.gyp` file, which describes the configuration
-to build your module in a JSON-like format. This file gets placed in the root of
-your package, alongside the `package.json` file.
+A `binding.gyp` file describes the configuration to build your module, in a
+JSON-like format. This file gets placed in the root of your package, alongside
+`package.json`.
 
-A barebones `gyp` file appropriate for building a node addon looks like:
+A barebones `gyp` file appropriate for building a Node.js addon coud look like:
```
Comment (richardlau) [inline README.md]: `coud`->`could`

**Comment by maclover7** [comment]
updated @richardlau

**Comment by maclover7** [comment]
CI: https://ci.nodejs.org/job/nodegyp-test-pull-request/56/

**Comment by maclover7** [comment]
Landed in c5e9802

**Pull request merged**